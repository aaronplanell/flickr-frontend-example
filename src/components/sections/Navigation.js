import React from 'react';
import { DEFAULT_VIEW_SIZE } from '../../config/constants';

const styleOfList = {
  paddingTop: '10px',
  textAlign: 'center'
}

const Navigation = ({selectViewSize, currentViewSize}) => (
  <nav>
    <ul>
      <li style={styleOfList}>
        <a href="/" className="btn">Albums</a>
      </li>
      <li style={styleOfList}>
        <a href="/about" className="btn">About</a>
      </li>
      <li style={styleOfList}>
        <input type="range"
          min={DEFAULT_VIEW_SIZE - 2}
          max={DEFAULT_VIEW_SIZE + 2}
          value={currentViewSize}
          onChange={event => selectViewSize(parseInt(event.target.value, 0))}
        />
      </li>
    </ul>
  </nav>
);

export default Navigation;

Navigation.propTypes = {
  selectViewSize: React.PropTypes.func.isRequired,
  currentViewSize: React.PropTypes.number
}
