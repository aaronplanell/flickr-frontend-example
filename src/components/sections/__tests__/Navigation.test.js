import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Navigation from '../Navigation';

const currentViewSize = 4;
const selectViewSize = jest.fn();

it('Navigation Renders OK', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <Navigation
      currentViewSize={currentViewSize}
      selectViewSize={selectViewSize}
    />,
    div);
});

it('Navigation Snapshot matchs', () => {
  const tree = renderer.create(
    <Navigation
      currentViewSize={currentViewSize}
      selectViewSize={selectViewSize}
    />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
