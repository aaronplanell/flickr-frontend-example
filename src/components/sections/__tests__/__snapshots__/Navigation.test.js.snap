exports[`test Navigation Snapshot matchs 1`] = `
<nav>
  <a
    className="btn"
    href="/">
    Albums
  </a>
  <a
    className="btn"
    href="/about">
    About
  </a>
  <input
    max={6}
    min={2}
    onChange={[Function]}
    type="range"
    value={4} />
</nav>
`;
